alacritty/src/daemon.rs

```diff
@@ -59,7 +59,12 @@ where
     let mut command = Command::new(program);
     command.args(args).stdin(Stdio::null()).stdout(Stdio::null()).stderr(Stdio::null());
     if let Ok(cwd) = foreground_process_path(master_fd, shell_pid) {
```

Comment:
I think the following should work.

```suggestion
    if let Some(cwd) = foreground_process_path(master_fd, shell_pid).ok().filter(Path::is_dir) {
```

Reply:
It didn't compile, but with the closure it did:

```
   Compiling alacritty v0.16.0-dev (/home/tycho/packages/alacritty/alacritty)
error[E0631]: type mismatch in function arguments
  --> alacritty/src/daemon.rs:65:82
   |
65 | ...id).ok().filter(Path::is_dir) {
   |             ------ ^^^^^^^^^^^^
   |             |      |
   |             |      expected due to this
   |             |      found signature defined here
   |             required by a bound introduced by this call
   |
   = note: expected function signature `for<'a> fn(&'a PathBuf) -> _`
              found function signature `fn(&Path) -> _`
note: required by a bound in `std::option::Option::<T>::filter`
  --> /rustc/9fc6b43126469e3858e2fe86cafb4f0fd5068869/library/core/src/option.rs:1477:5
help: consider wrapping the function in a closure
   |
65 |     if let Some(cwd) = foreground_process_path(master_fd, shell_pid).ok().filter(|arg0: &PathBuf| Path::is_dir(/* &Path */)) {
   |                                                                                  ++++++++++++++++             +++++++++++++

For more information about this error, try `rustc --explain E0631`.
error: could not compile `alacritty` (bin "alacritty") due to 1 previous error
```